boa/src/syntax/ast/node/array/mod.rs

```diff
@@ -39,14 +39,26 @@ impl Executable for ArrayDecl {
         for elem in self.as_ref() {
             if let Node::Spread(ref x) = elem {
                 let val = x.run(interpreter)?;
-                let mut vals = interpreter.extract_array_properties(&val).unwrap();
-                elements.append(&mut vals);
-                continue; // Don't push array after spread
+                let iterator_record =
+                    super::super::super::super::builtins::iterable::get_iterator(interpreter, val)?;
```

Comment:
```suggestion
                let iterator_record = iterable::get_iterator(interpreter, val)?;
```
It's better if we import the iterable module on top.